lib/miq_automation_engine/models/miq_git_repo.rb

```diff
@@ -0,0 +1,60 @@
+class MiqGitRepo < ApplicationRecord
+  include AuthenticationMixin
+  serialize :branches
+  serialize :tags
+  validates_presence_of :url
+  validates :url, :format => URI::regexp(%w(http https))
+  before_save :default_values
+
+  def verify_credentials(_auth_type = nil, _options = {})
+    refresh
```

Comment:
This doesn't feel right (that we would verify *and* it has a side effect of collecting data)